pyrealm/demography/crown.py

```diff
@@ -228,6 +139,38 @@ def calculate_relative_canopy_radius_at_z(
     return m * n * z_over_height ** (n - 1) * (1 - z_over_height**n) ** (m - 1)
 
 
+def calculate_crown_radius(
+    q_z: NDArray[np.float32],
+    r0: NDArray[np.float32],
+    validate: bool = True,
+) -> NDArray[np.float32]:
+    r"""Calculate crown radius from relative crown radius and crown r0.
+
+    The relative crown radius (:math:`q(z)`) at a given height :math:`z` describes the
+    vertical profile of the crown shape, but only varies with the ``m`` and ``n`` shape
+    parameters and the stem height. The actual crown radius at a given height
+    (:math:`r(z)`) needs to be scaled using :math:`r_0` such that the maximum crown area
+    equals the expected crown area given the crown area ratio traiit for the plant
+    functional type:
+
+    .. math::
+
+        r(z) = r_0 q(z)
+
+    This function calculates :math:`r(z)` given estimated ``r0`` and an array of
+    relative radius values.
+
+    Args:
+        q_z: TODO
```

Comment:
To do?

Reply:
Hmm - to be more precise about once I've thought through dimensionality. But I can make a start on the description for now!